src/icq/system.rs

```diff
@@ -125,11 +125,12 @@ impl ICQSystem {
             let session_info = protocol::start_session(&registered_account_info).await;
             log::debug!("Session info: {:?}", session_info);
             match session_info {
-                Ok(_info) => {
+                Ok(session) => {
                     self.tx
                         .connection_proxy(&handle)
                         .set_state(purple::PurpleConnectionState::PURPLE_CONNECTED)
                         .await;
+                    *account_info.protocol_data.lock().await = Some(AccountData { session });
```

Comment:
What does this do?

Reply:
- `account_info.protocol_data` is a `Mutex<Option<AccountData>>`
- `.lock()` get a lock on the mutex so we can have a mutable reference to the Option<AccountData>. Also, `.lock()` actually return a `Future<LockGuard>` 
- `.await` await that future
- `*` deref the ScopedGuard into our `Option<AccountData>`